### PR TITLE
Sprint 47: TLT-1314 and TLT-1979: crosslisting and decrosslisting frontend

### PR DESCRIPTION
**Pull request by elliottyates**
Frontend and proxy layer udpates to support TLT-1314 (new xlist map) and TLT-1979 (delete xlist map).

Split (non-merged) stories are in:
https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/tree/task/elliottyates/tlt-1314/crosslist_courses
https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/tree/task/elliottyates/tlt-1979/decrosslisting

### CODE REVIEW

File: cross_list_courses/static/cross_list_courses/js/controllers/ListController.js [hunk 1]
```diff
@@ -3,103 +3,272 @@
...
     var app = angular.module('CrossListCourses');
-    app.controller('ListController',
-        ['$scope', '$http', '$timeout', '$document', '$window', '$compile','djangoUrl','$log',
-        function($scope, $http, $timeout, $document, $window, $compile, $djangoUrl, $log){
+    app.controller('ListController', ListController);
 
-            $scope.dtInstance = null;
+    function ListController($scope, $http, $timeout, $document, $window,
+                            $compile, djangoUrl, $log, $q, $uibModal) {
+        // {alertType: 'bootstrap alert class', text: 'actual message'}
```
Comment (sapnamysore) [inline cross_list_courses/static/cross_list_courses/js/controllers/ListController.js]: This can be deleted?

**Comment by eparker71** [comment]
👍 Looks good

File: cross_list_courses/static/cross_list_courses/js/controllers/ListController.js (hunk 1 above)
Comment (elliottyates) [inline cross_list_courses/static/cross_list_courses/js/controllers/ListController.js]: That's meant to indicate the expected `$scope.message` format -- I can update the code to make that clearer.

Comment (sapnamysore) [inline cross_list_courses/static/cross_list_courses/js/controllers/ListController.js]: Cool

**Comment by sapnamysore** [comment]
Looks good to me too! 👍

**Pull request merged**